### PR TITLE
🐳 Containerize the CI pipeline

### PR DESCRIPTION
**Pull request by TOTBWF**
## Patch Description

This PR changes our CI from using the https://github.com/ocaml/setup-ocaml github action to a bespoke action based off of docker containers. This allows us to better cache our dependencies, which reduces our CI run times from ~3mins to ~1.5 mins.

## Notes
We were running the CI on both pull_request and push, which means we had 2 duplicate CI runs for open PRs. I've changed this to just be on push, as this reduces some noise and keeps our cache smaller.

### CODE REVIEW

File: test/Test.ml [hunk 1]
```diff
@@ -4,7 +4,7 @@ let header fname =
   String.make 20 '-' ^ "[" ^ fname ^ "]" ^ String.make 20 '-' ^ "\n"
 
 let execute_file fname =
-  if String.equal (Filename.extension fname) ".cooltt" then
+  if String.equal (Filename.extension fname) ".cooltt" && fname <> "META.cooltt" then
```
Comment (favonia) [inline test/Test.ml:7]: Taste it! I don't know why it's triggered in Docker.

Comment (jonsterling) [reply test/Test.ml:7]: 🥴🤦🏻‍♂️

Comment (favonia) [reply test/Test.ml:7]: Feel free to add/suggest OCaml comments for non-genie mortals.

**Review by favonia — APPROVED** [review]
I'm done. :rocket:

**Pull request merged**